crates/cargo-wdk/src/actions/package/mod.rs

```diff
@@ -0,0 +1,460 @@
+// Copyright (c) Microsoft Corporation
+// License: MIT OR Apache-2.0
+//! This module contains the `PackageAction` struct and its associated methods
+//! for orchestrating the packaging of a driver project. It includes the build
+//! step as a prerequisite for packaging. It consists the logic to build and
+//! package standalone projects, workspaces, individual members in a workspace
+//! and emulated workspaces. It handles various tasks such as creation of the
+//! `PackageTask` struct and interacting with `wdk-build`.
+
+#[cfg(test)]
+mod tests;
+
+mod error;
+use cargo_metadata::{Metadata as CargoMetadata, Package, TargetKind};
+use error::PackageActionError;
+use mockall_double::double;
+mod package_task;
+
+use std::{
+    fs::read_dir,
+    io,
+    path::{Path, PathBuf},
+    result::Result::Ok,
+};
+
+use anyhow::Result;
+use package_task::{PackageTask, PackageTaskParams};
+use tracing::{debug, error as err, info, warn};
+use wdk_build::{
+    metadata::{TryFromCargoMetadataError, Wdk},
+    CpuArchitecture,
+};
+
+use crate::actions::{build::BuildAction, Profile};
+#[double]
+use crate::providers::{exec::CommandExec, fs::Fs, metadata::Metadata, wdk_build::WdkBuild};
+
+pub struct PackageActionParams<'a> {
+    pub working_dir: &'a Path,
+    pub profile: Option<&'a Profile>,
+    pub target_arch: Option<&'a CpuArchitecture>,
+    pub verify_signature: bool,
+    pub is_sample_class: bool,
+    pub verbosity_level: clap_verbosity_flag::Verbosity,
+}
+
+/// Action that orchestrates the packaging of a driver project
+/// This also includes the build step as pre-requisite for packaging
+pub struct PackageAction<'a> {
+    working_dir: PathBuf,
+    profile: Option<&'a Profile>,
+    target_arch: Option<&'a CpuArchitecture>,
```

Comment:
Nit: probably better to use `CpuArchitecture` instead of `&CpuArchitecture` as it's a simple enum.

Reply:
It's a super small change. Might as well make it.